packages/blockly/src/blocks/block-factory.ts

```diff
@@ -326,15 +326,30 @@ export function registerCustomBlocks(customBlocks: ICustomBlock[]) {
         // For now, though, we just return a simple create command.
         const count = block.getFieldValue("count");
         const type = block.getFieldValue("type").toLowerCase().replace(/\s+/g, "_");
-        const statements = netlogoGenerator.statementToCode(block, "statements");
+        const statements = javascriptGenerator.statementToCode(block, "statements");
+        const callback = statements ? `, (agent) => {\n${statements}\n}` : "";
 
-        return `create-${type} ${count}\n${statements}`;
+        return `create_${type}(${count}${callback});\n`;
       } else if (blockDef.type === "ask") {
         const target = block.getFieldValue("target");
-        const targetEntity = blockConfig.targetEntity || "";
-        const statements = netlogoGenerator.statementToCode(block, "statements");
-        return `ask ${target} ${targetEntity} [\n${statements}]\n`;
+        const statements = javascriptGenerator.statementToCode(block, "statements");
+        const agents = target === "all" ? "sim.actors" : `sim.withLabel("${target}")`;
+        return `${agents}.forEach(agent => {\n${statements}\n});\n`;
       } else if (blockDef.type === "condition") {
+        if (blockConfig.generatorTemplate) {
+          let code = String(blockConfig.generatorTemplate);
+          // Replace ${PARAM} placeholders with field values
+          const params = Array.isArray(blockConfig.parameters) ? blockConfig.parameters : [];
+          params.forEach((p: IParameter) => {
+            const val = block.getFieldValue(p.name);
+            const safe = val != null ? String(val) : "";
+            const re = new RegExp(`\\$\\{${p.name}\\}`, "g");
+            code = code.replace(re, safe);
+          });
```

Comment:
Should this be moved into a helper function that can also be used by the similar code in the action block section above (lines 290-295)?